docs/sources/mimir/manage/tools/mimirtool.md

```diff
@@ -158,6 +158,56 @@ The following command deletes the Alertmanager configuration in the Grafana Mimi
 mimirtool alertmanager delete
 ```
 
+#### Prepare Alertmanager configuration for UTF-8 in Alertmanager 0.27 and later
+
+Starting with Alertmanager v0.27.0, Alertmanager has a new parser for matchers to support UTF-8
+characters in Alertmanager, and this parser has a number of backwards incompatible changes. While
+most matchers will be forward-compatible, some will not. Alertmanager is operating a transition
+period where it supports both UTF-8 and classic matchers, and your configuration will continue
+to work in Mimir until your Mimir installation enables UTF-8 strict mode. All Mimir installations
+should enable UTF-8 strict mode before the end of the transition period, so we recommend you use
+the prepare command to prepare your Alertmanager configuration for this change.
+
+The prepare command automatically translates any matchers that are incompatible with the UTF-8
+parser into equivalent matchers that are compatible by enforcing double quoting on the right hand side
+of the matcher. This does not change their behavior, rather UTF-8 matchers require double quoting where
+whitespace and reserved characters are present. However, it's much easier to enforce double quoting
+on all matchers instead of just the few that are incompatible.
+
+The command takes as input an existing configuration and template files, and prints as output
+the prepared configuration and template files:
+
+```bash
+mimirtool alertmanager prepare <config_file> [template_files...]
+```
+
+You can also output the prepared configuration and template files to a folder which can be
+reviewed before being loaded back into the Alertmanager at a later time. It is recommended to output
+the prepared files to a different folder than the original files so you always have the original files
+as a backup. For example, the following command outputs the prepared configuration and template files
+to a folder called `prepared`:
+
+```bash
+mimirtool alertmanager prepare <config_file> [template_files...] --output-dir="prepared"
+```
+
+Within the output dir, the configuration file is named `config.yaml` and the template files ends in
+`.tpl`, where each template is written out to its own file. Note that using the `--output-dir` flag
+only writes the output to files and no longer print the config to the console.
+
+Once your configuration has been prepared, verify it using the verify command and the
+`--utf8-strict-mode` flag:
+
+```bash
+mimirtool alertmanager verify <config_file> [template_files...] --utf8-strict-mode
+```
+
+You should see a warning `UTF-8 mode enabled` to let you know that UTF-8 strict mode was enabled
+when validating your configuration.
+
+If the command exits without error, and you are satisfied with the changes made to the prepared
+configuration and template files, reload it using the `load` command.
```

Comment:
I've rewritten the documentation to adhere to the Grafana Labs style a little more closely, avoiding passive voice and using the [`admonition` shortcode](https://grafana.com/docs/writers-toolkit/write/shortcodes/#admonition).

I've also used [semantic line breaks](https://sembr.org/) in the suggestion which I believe are better for future line based diffing in the GitHub UI and result in the same HTML output.

I've removed the entirety of the first paragraph and the references to the Alertmanager version and transition period. I think this information is really important but is best announced in release notes or changelogs and then the documentation can remain timeless.

I'm not sure where exactly the announcement should go for the Grafana Mimir project, hopefully another maintainer can help with that.

```suggestion
#### Prepare Alertmanager configuration for UTF-8

The prepare command translates any matchers that are incompatible with the Alertmanager UTF-8 parser into equivalent matchers that are.
The translation doesn't change the behavior of the matcher.

The command takes as input an Mimir configuration file and template files, and prints as output the prepared configuration and template files:

```bash
mimirtool alertmanager prepare <config_file> [template_files...]
```

You can also output the prepared configuration and template files to a directory which you can review before loading it into Alertmanager at a later time.
You should output the prepared files to a different directory than the original files so you always have the original files as a backup.

For example, the following command outputs the prepared configuration and template files to a directory called `prepared`:

```bash
mimirtool alertmanager prepare <config_file> [template_files...] --output-dir=prepared
```

The command writes each prepared template to its own file, with a `.tpl` suffix and the configuration file to `config.yaml`.

{{< admonition type="note" >}}
When you use the `--output-dir` flag, the command only writes the output to files and doesn't print the configuration to the console.
{{< /admonition >}}

After you have prepared your configuration, you can verify it using the `verify` command and the `--utf8-strict-mode` flag:

```bash
mimirtool alertmanager verify <config_file> [template_files...] --utf8-strict-mode
```

The command outputs a warning containing the text `UTF-8 mode enabled` that lets you know that command is using UTF-8 strict mode to validate your configuration.

If the command exits without error, and you're satisfied with the changes made to the prepared configuration and template files, reload it using the `load` command.
```

Reply:
Looks good! Let's do the semantic lines breaks to the entire file in a separate PR, otherwise it will be super inconsistent?

Reply:
Sure, incorporate the suggestion however you please :)